contracts/src/challengeV2/EdgeChallengeManager.sol

```diff
@@ -396,25 +429,24 @@ contract EdgeChallengeManager is IEdgeChallengeManager, Initializable {
                 claimStateData.executionState
             );
 
-            edgeAdded = store.createLayerZeroEdge(args, ard, oneStepProofEntry, expectedEndHeight, NUM_BIGSTEP_LEVEL);
+            edgeAdded = store.createLayerZeroEdge(
+                args, ard, oneStepProofEntry, expectedEndHeight, NUM_BIGSTEP_LEVEL, initialStakeAmount, stakeAmountSlope
+            );
         } else {
-            edgeAdded = store.createLayerZeroEdge(args, ard, oneStepProofEntry, expectedEndHeight, NUM_BIGSTEP_LEVEL);
+            edgeAdded = store.createLayerZeroEdge(
+                args, ard, oneStepProofEntry, expectedEndHeight, NUM_BIGSTEP_LEVEL, initialStakeAmount, stakeAmountSlope
+            );
         }
 
         IERC20 st = stakeToken;
-        uint256 sa = stakeAmount;
         // when a zero layer edge is created it must include stake amount. Each time a zero layer
         // edge is created it forces the honest participants to do some work, so we want to disincentive
         // their creation. The amount should also be enough to pay for the gas costs incurred by the honest
         // participant. This can be arranged out of bound by the excess stake receiver.
         // The contract initializer can disable staking by setting zeros for token or amount, to change
         // this a new challenge manager needs to be deployed and its address updated in the assertion chain
-        if (address(st) != address(0) && sa != 0) {
-            // since only one edge in a group of rivals can ever be confirmed, we know that we
-            // will never need to refund more than one edge. Therefore we can immediately send
-            // all stakes provided after the first one to an excess stake receiver.
-            address receiver = edgeAdded.hasRival ? excessStakeReceiver : address(this);
-            st.safeTransferFrom(msg.sender, receiver, sa);
+        if (address(st) != address(0) && edgeAdded.stakeAmount != 0) {
+            st.safeTransferFrom(msg.sender, address(this), edgeAdded.stakeAmount);
```

Comment:
we should allow some slippage control on the stakeAmount transferred, although this technically can be done by limiting approval it might make sense to have an extra parameter?

Reply:
checked in this call
https://github.com/OffchainLabs/bold/blob/28823e3da73c11bd8ef3762f30be6a22247ad93d/contracts/src/challengeV2/libraries/EdgeChallengeManagerLib.sol#L442